docs/basics/101-125-config.rst

```diff
@@ -0,0 +1,459 @@
+   section :ref:`nesting`. When you install a DataLad dataset
+   as a subdataset, it gets *registered* in this file.
+   For each subdataset, an individual submodule entry
+   will store the information about the subdatasets
+   ``--source`` or *origin* (the "url").
+   Thus, every subdataset (and sub-subdataset, and so forth) in your dataset
+   will be listed in this file.
+   If you want, go back to section :ref:`installds` to see that the
+   "url" is the same URL we installed the longnow dataset from, and
+   go back to section :ref:`sharelocal1` to remind yourself of
+   how installing a dataset with subdatasets looked and felt like.
+
+   Another interesting part is the last section, "remote".
+   Here we can find the :term:`sibling` "roommate" we defined
+   in :ref:`sibling`. The term :term:`remote` is Git-terminology and is
+   used to describe other repositories or DataLad datasets that the
+   repository knows about and tracks.
+   This file, therefore, is where DataLad *registered* the sibling
+   with :command:`datalad siblings add`, and thanks to it you can
+   collaborate with your room mate.
+   Note the *path* given as a value to the ``url`` variable. If at any point
+   either your superdataset or the remote moves on your file system,
+   the association between the two datasets breaks -- this can be fixed by adjusting this
+   path, and a demonstration of this is in section (Todo: link file system management).
+   `fetch` contains a specification which parts of the repository are
+   updated - in this case the everything (all of the branches).
+   Lastly, the ``annex-ignore = false`` configuration allows Git-annex
+   to query the remote when it tries to retrieve data from annexed content.
+
+``.git/config`` versus other configuration files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Within a Git repository, ``.git/config`` is not the only configuration file.
+There are also ``.gitmodules`` and ``.gitattributes``, and in DataLad datasets
+there also is a ``.datalad/config`` file.
+
+One crucial aspect distinguishes the ``.git/config`` file from those other files:
+*It will not be shared.* The reason for this is simple. This config file
+lives in the ``.git`` directory of your repository, and this directory is not
+version controlled
+
+
+.. todo::
+
+   * understand why there are .gitmodules files with the same information
+     as in the .gitmodules section, and also redundancy to .gitattributes (the
+     git annex backend. which files need to be modified to change a specific
+     option?
+
+   * .gitmodules is a configuration file that stores the mapping between
+     the projects URL and the local subdirectory you've pulled it into.
+     There will be an entry for each submodule (subdataset) in your dataset.
+     This file is how other people that install your superdataset know where to get
+     submodule projects from.
+
+   * To change the backend, you should set it in ``.gitattributes``
```

Comment:
FTR: To really switch everything to a new backend, you need to use the `git annex migrate` command, otherwise only new files are affected. It is not problem to have different keys use different backends.